### DISCUSSION TITLE
Getting following error while gathering metrics from multiple services

### DISCUSSION THREAD

**Discussion by ychaydhari** [open-telemetry/opentelemetry-java-instrumentation#6763] [Q&A]
Hello,
When I started two services the exporter failed to gather metrics.

command line:- 
`-javaagent:"C:\Workspace\api\Service\opentelemetry-javaagent.jar" -Dotel.metrics.exporter=otlp -Dotel.exporter.otlp.metrics.endpoint=http://localhost:4317 -Dotel.traces.exporter=zipkin -Dotel.metrics.exemplar.filter=none -Dotel.service.name=admin_service`

`-javaagent:"C:\Workspace\MVIS_2.0\mvis\cm\api\ConnectionManagerService\opentelemetry-javaagent.jar" -Dotel.metrics.exporter=otlp -Dotel.exporter.otlp.metrics.endpoint=http://localhost:4317 -Dotel.traces.exporter=zipkin -Dotel.metrics.exemplar.filter=none -Dotel.service.name=cm_service`

Initially, I'm getting metrics from both the services when I hit API of cm_service it throws an exception.

**Comment by ychaydhari** [comment]
Exception :- 

`2022-09-27T15:17:21.338+0530    info    MetricsExporter {"kind": "exporter", "data_type": "metrics", "name": "logging", "#metrics": 20}
2022-09-27T15:17:21.839+0530    error   prometheusexporter@v0.59.0/log.go:34    error gathering metrics: collected metric "http_server_active_requests" { label:<name:"http_flavor" value:"1.1" > label:<name:"http_host" value:"localhost:7171" > label:<name:"http_method" value:"GET" > label:<name:"http_scheme" value:"http" > label:<name:"job" value:"cm_service" > gauge:<value:-5 > } was collected before with the same name and label values
        {"kind": "exporter", "data_type": "metrics", "name": "prometheus"}
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter.(*promLogger).Println
        github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter@v0.59.0/log.go:34
github.com/prometheus/client_golang/prometheus/promhttp.HandlerForTransactional.func1
        github.com/prometheus/client_golang@v1.13.0/prometheus/promhttp/http.go:139
net/http.HandlerFunc.ServeHTTP
        net/http/server.go:2084
net/http.(*ServeMux).ServeHTTP
        net/http/server.go:2462
go.opentelemetry.io/collector/config/confighttp.(*decompressor).wrap.func1
        go.opentelemetry.io/collector@v0.59.0/config/confighttp/compression.go:162
net/http.HandlerFunc.ServeHTTP
        net/http/server.go:2084
go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.(*Handler).ServeHTTP
        go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.34.0/handler.go:204
go.opentelemetry.io/collector/config/confighttp.(*clientInfoHandler).ServeHTTP
        go.opentelemetry.io/collector@v0.59.0/config/confighttp/clientinfohandler.go:39
net/http.serverHandler.ServeHTTP
        net/http/server.go:2916
net/http.(*conn).serve
        net/http/server.go:1966`

**Reply by ychaydhari** [reply]
Configure pipeline :- 

service:
  extensions: [health_check]
  pipelines:
    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, prometheus]
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, zipkin]`

**Reply by laurit** [reply]
As this error seems to be reported by the collector it might be better to ask this question from the collector folks.

**Reply by ychaydhari** [reply]
yes, because I configuration to drop this `http_server_active_requests` metric but still getting these metrics.
